packages/math/src/SphericalHarmonics3.ts

```diff
@@ -0,0 +1,242 @@
+import { IClone } from "@oasis-engine/design";
+import { Color } from "./Color";
+import { Vector3 } from "./Vector3";
+
+/**
+ * Use SH3 to represent irradiance environment maps efficiently, allowing for interactive rendering of diffuse objects under distant illumination.
+ * @remarks
+ * https://graphics.stanford.edu/papers/envmap/envmap.pdf
+ * http://www.ppsloan.org/publications/StupidSH36.pdf
+ * https://google.github.io/filament/Filament.md.html#annex/sphericalharmonics
+ */
+export class SphericalHarmonics3 implements IClone {
+  static fromArray(shArray: number[], out: SphericalHarmonics3): void {
```

Comment:
You can use setValueByArray likce `Vector3`,`Vector2`.

Reply:
+1